app/components/questions/TaxiPaymentMethod.js

```diff
@@ -0,0 +1,115 @@
+import { useState, useEffect } from 'react'
+import useFetch from 'use-http'
+import { Table } from '../../components/Table'
+import { Notification } from '../../components/Notification'
+import {
+  Formatter
+} from '../../site/metrics'
+
+const COMMON_PAYMENT_ENDPOINT = `${process.env.NEXT_PUBLIC_API}/question/taxitrips`;
+
+const COMMON_PPAYMENT_COLS = [
+  {
+    key: "pickup_community_area",
+    name: "Pickup Location",
+  },
+  {
+    key: "payment_type",
+    name: "Payment Method",
+    rowClasses: ["center"],
+  },
+  {
+    key: "max_count",
+    name: "# of Payments",
+    rowClasses: ["center"],
+    rgb: "255, 165, 0",
+    alpha: (v) => (v/450),
+  },
+  {
+    key: "percentage",
+    name: "% from Total Taxi Rides",
+    format: Formatter.percentWithOneDecimal,
+    rowClasses: ["center"],
+    rgb: "136, 34, 85",
+    alpha: (v) => (v / .7),
+  },
+];
+
+const COMMON_DPAYMENT_COLS = [
+  {
+    key: "dropoff_community_area",
+    name: "Dropoff Location",
+  },
+  {
+    key: "payment_type",
+    name: "Payment Method",
+    rowClasses: ["center"],
+  },
+  {
+    key: "max_count",
+    name: "# of Payments",
+    rowClasses: ["center"],
+    rgb: "255, 165, 0",
+    alpha: (v) => (v/450),
+  },
+  {
+    key: "percentage",
+    name: "% from Total Taxi Rides",
+    format: Formatter.percentWithOneDecimal,
+    rowClasses: ["center"],
+    rgb: "136, 34, 85",
+    alpha: (v) => (v / .7),
+  },
+];
+
+function transformData(res) {
+  if (res) {
+    const payment_per_pickup = res.payment_per_pickup;
+    const payment_per_dropoff = res.payment_per_dropoff;
+    return [ 
+      payment_per_pickup,
+      payment_per_dropoff
+      ];
+  }
+  return [ [] , [] ];
+}
+
+export default function TaxiTrips(props) {
+  const { loading, error, data } = useFetch(COMMON_PAYMENT_ENDPOINT, {}, []);
+  const [ payment_per_pickup, payment_per_dropoff ] = transformData(data);
```

Comment:
Since all `transformData()` does is pull out these two lists, you could also write it like this:

```js
const { loading, error, data } = useFetch(COMMON_PAYMENT_ENDPOINT, {}, []);
const payment_per_pickup = data?.payment_per_pickup || [];
const payment_per_dropoff = data?.payment_per_dropoff || [];
```

The syntax `?.` is "optional chaining," if `data` is undefined or null, instead of throwing an error, it will return undefined.

The syntax `||` means "or" when you use it in a conditional, but it has another use:

```js
const something = ???
const output = something || "default";
```

If `something` is null or undefined, the `||` will cause the expression to evaluate to `"default"`.

What you have works, just wanted to explain some JavaScript syntax that may be new and handy to you.